libs/runtime/ebpf_hash_table.c

```diff
@@ -57,17 +57,15 @@ struct _ebpf_hash_table
     size_t bucket_count_mask; // Mask to use to get bucket index from hash.
     volatile size_t
         entry_count; // Count of entries in the hash table. Only valid if max_entry_count != EBPF_HASH_TABLE_NO_LIMIT.
-    size_t max_entry_count;         // Maximum number of entries allowed or EBPF_HASH_TABLE_NO_LIMIT if no maximum.
-    uint32_t seed;                  // Seed used for hashing.
-    size_t key_size;                // Size of key.
-    size_t value_size;              // Size of value.
-    size_t supplemental_value_size; // Size of supplemental value.
-    void* (*allocate)(size_t size); // Function to allocate memory.
-    void (*free)(void* memory);     // Function to free memory.
-    void (*extract)(
-        _In_ const uint8_t* value,
-        _Outptr_ const uint8_t** data,
-        _Out_ size_t* num); // Function to extract bytes to hash from key.
+    size_t max_entry_count;            // Maximum number of entries allowed or EBPF_HASH_TABLE_NO_LIMIT if no maximum.
+    uint32_t seed;                     // Seed used for hashing.
+    size_t key_size;                   // Size of key.
+    size_t value_size;                 // Size of value.
+    size_t supplemental_value_size;    // Size of supplemental value.
+    ebpf_hash_table_allocate allocate; // Function to allocate memory.
+    ebpf_hash_table_free free;         // Function to free memory.
+    ebpf_hash_table_extract_function extract; // Function to extract bytes to hash from key.
+    uint32_t allocate_tag;                    // Pool tag to use for allocations.
```

Comment:
I think `allocation_tag` would be better grammatically, since something starting with a verb sounds like a function pointer (like "allocate" in line 65 is).